pkg/inputs/jetstream_input/jetstream_input.go

```diff
@@ -190,13 +199,42 @@ func (n *jetstreamInput) workerStart(ctx context.Context) error {
 		return fmt.Errorf("failed to get stream: %v", err)
 	}
 
+	// Get stream info to determine retention policy
+	streamInfo, err := s.Info(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to get stream info: %v", err)
+	}
+
+	// Determine ack policy and deliver policy based on stream retention
+	// Workqueue streams have specific requirements
+	ackPolicy := jetstream.AckAllPolicy
+	deliverPolicy := toJSDeliverPolicy(n.Cfg.DeliverPolicy)
+
+	if streamInfo.Config.Retention == jetstream.WorkQueuePolicy {
+		// Workqueue streams require explicit ack
+		ackPolicy = jetstream.AckExplicitPolicy
+		// Workqueue streams require deliver all policy
+		deliverPolicy = jetstream.DeliverAllPolicy
+	}
+
+	// Determine filter subjects based on consumer mode
+	var filterSubjects []string
+	switch n.Cfg.ConsumerMode {
+	case consumerModeSingle:
+		// Use configured subjects as filter
+		filterSubjects = n.Cfg.Subjects
+	case consumerModeMulti:
+		// Use explicitly configured filter-subjects
+		filterSubjects = n.Cfg.FilterSubjects
+	}
+
```

Comment:
When `ConsumerMode` and `FilterSubjects` are removed we should just use Subjects in `CreateOrUpdateConsumer`